include/envoy/ssl/context_config.h

```diff
@@ -111,6 +111,12 @@ class ContextConfig {
    * @return The maximum TLS protocol version to negotiate.
    */
   virtual unsigned maxProtocolVersion() const PURE;
+
+  /**
+   * @return true if the config is valid. Only when SDS dynamic secret is needed, but has not been
+   * downloaded yet, the config is invalid.
+   */
+  virtual bool isValid() const PURE;
```

Comment:
Would `isReady` be better name for this?

Reply:
+1